test/intl402/Temporal/PlainMonthDay/from/reference-year-1972.js

```diff
@@ -44,28 +50,41 @@ TemporalHelpers.assertPlainMonthDay(
 
 const result5 = Temporal.PlainMonthDay.from({ year: 5781, monthCode: "M02", day: 30, calendar: "hebrew" });
 TemporalHelpers.assertPlainMonthDay(
-  result5, "M02", 30,
-  "year is ignored if monthCode is given (Cheshvan 5781 has 29 days)",
-  1971
+  result5, "M02", 29,
+  "year and monthCode determine if calendar date exists, and reference year must agree (Cheshvan 5781 has 29 days)",
+  1972
 );
 
 const result6 = Temporal.PlainMonthDay.from({ year: 5781, month: 2, day: 30, calendar: "hebrew" }, { overflow: "constrain" });
 TemporalHelpers.assertPlainMonthDay(
   result6, "M02", 29,
-  "if monthCode is not given, year is used to determine if calendar date exists, but reference year still correct",
+  "year and month determine if calendar date exists, and reference year must agree (Cheshvan 5781 has 29 days)",
   1972
 );
```

Comment:
For the Gregorian calendar and Hebrew calendar these looked good. I do not have a deep enough background on temporal to know if other calendars would be affected.

I checked a Hebrew calendar on both pr's to make sure the dates mentioned were accurate. I used this calendar to check - https://www.chabad.org/calendar/view/year.asp?tdate=1/1/2021

<img width="1205" alt="Screenshot 2024-09-19 at 10 55 24 AM" src="https://github.com/user-attachments/assets/6d3ac27a-ed4e-4a55-a390-38e9b6695dc0">

<img width="1210" alt="Screenshot 2024-09-19 at 10 55 34 AM" src="https://github.com/user-attachments/assets/5c8c5609-0e29-4888-ac50-4a9861239d80">